src/common/Log/Log.h

```diff
@@ -0,0 +1,80 @@
+//
+// Created by XFC on 2018/6/24.
+//
+
+#ifndef SNAIL_LOGCONTEXT_H
+#define SNAIL_LOGCONTEXT_H
+
+#include <list>
+#include <functional>
+#include <sstream>
+#include <iostream>
+#include <vector>
+#include <map>
+#include <thread>
+
+namespace snail {
+    namespace log {
+        using LogHandler = std::function<void(const std::string &msg)>;
+        enum Loglevel {
+            Debug = 0,
+            Trace,
+            Warn,
+            Error
+        };
+
+        class LogContext {
+        public:
+            LogContext() : m_cache_(), m_currCacheSize_(0), m_maxCacheSize_(0) {}
+            virtual ~LogContext() = default;
+
+        public:
+            void addHandler(const LogHandler &handler, bool enableCache);
+            void flush(const std::string &msg);
+            void setCacheSize(int size);
+
+        private:
+            std::list<LogHandler> m_noCacheFunc_;
```

Comment:
noCacheFuncs

Reply:
ok, I  will reslove this spell error.

Reply:
Not spelling lint error. I means that for a member, we should write it as a `m` prefix.